File: firmware/ghar_sajag_reference_v1_3/backend/ghar_sajag/feature_flags.py
"""Build/deployment feature gates. P1 capabilities are deny-by-default."""
from __future__ import annotations
import os

DEFAULTS = {
    "morning_routine": True, "call_family": True, "door_history": True,
    "daily_summary": True, "local_offline": True, "temperature_context": False,
    "external_camera": False, "fall_detection": False, "professional_response": False,
}

def enabled(name: str) -> bool:
    if name not in DEFAULTS:
        return False
    raw = os.getenv("GS_FEATURE_" + name.upper(), "")
    return raw.lower() in {"1", "true", "yes", "on"} if raw else DEFAULTS[name]

def snapshot() -> dict[str, bool]:
    return {name: enabled(name) for name in DEFAULTS}

File: firmware/ghar_sajag_reference_v1_3/app/src/platform/feature_flags.mjs
/** Build-injected flags. P1/P2 capabilities remain off unless explicitly built in. */
const defaults = Object.freeze({
  morning_routine: true, call_family: true, door_history: true, daily_summary: true,
  local_offline: true, temperature_context: false, external_camera: false,
  fall_detection: false, professional_response: false,
});

const injected = globalThis.__GS_FEATURES__ ?? {};
export const featureFlags = Object.freeze(Object.fromEntries(
  Object.keys(defaults).map((name) => [name, typeof injected[name] === "boolean" ? injected[name] : defaults[name]])
));
export function isFeatureEnabled(name) { return featureFlags[name] === true; }
